src/envoy/http/mixer/filter.cc

```diff
@@ -189,26 +189,16 @@ void Filter::UpdateHeaders(
 
 FilterHeadersStatus Filter::encodeHeaders(HeaderMap& headers, bool) {
   ENVOY_LOG(debug, "Called Mixer::Filter : {} {}", __func__, state_);
-  ASSERT(state_ == Complete || state_ == Responded);
+  // Init state is possible if a filter prior to mixerfilter interrupts the
+  // filter chain
+  ASSERT(state_ == NotStarted || state_ == Complete || state_ == Responded);
```

Comment:
My suggestion is to remove this ASSERT.

Reply:
I think it's useful to know that Calling is not happening. That looks like the only invariant in the control flow here, that "Check must be called before proceeding". I removed all the other asserts.